### DISCUSSION TITLE
some suggestions on chain throughput

### DISCUSSION THREAD

**Discussion by leviok** [filecoin-project/lotus#3763] [General Q&A] [answered]
1. winning post task spend a lot of time. omit it will save long time consuming. is it necessary? is it important for security?

2. since lotus is a PoS chain, mining is base on plenty of messages. i think lotus need to support distributed sync. miners can handle the rest(like large machine  or more machines to do sync). user can use light-node or open-api. other full node without mining block can afford sync time more than miner. I don’t have time to have a deep understanding of block validation, now. so i don't know how much logic can be parallel(like some nonce issue?). but i think this maybe help. 

3. this suggestion sounds not good enough, but you can refer to. if sync is not good enough. mining base like use some selection. then mining can drop some block to choose a less messages head. miner who produce it will lose some block weight. but i think it's not a big problem, chain can work with some fork and aggregation finally. i think this can solve some problems caused by extremely high throughput.

hope a better filecoin.

**Comment by whyrusleeping** [comment, marked as answer]
(1) Winning post is pretty important for security, but i'm wondering if we can make it faster by reducing security (this is a question for @nicola )

(2) Light clients are high priority for us to implement, I actually expect to see early versions land in under a month

(3) this is really hard, because of fork protection logic. Miners who mine on two different chains can be slashed (for good reason!).

**Reply by leviok** [reply]
My English level limits my expression, I can only use short sentences to explain a lot of meanings, which is more difficult.
1.  i assume that winning post is necessary for security. then can wining post use the ticket from head - 1? i think it only related who have the right to produce block, will not affect other logic more. but this can make wining post parallel. we can do sync and post in the same time window. This change should only slightly affect security, but can greatly save time.

2. i'm sure you are doing on it. i'm keeping an eye on dev channel in slack.  but i think filecoin faces unprecedented throughput challenges than other chains. so i still think we need some distributed sync program.

3. This is just an idea. I don’t know much about it. If you think it’s not feasible at all, then don’t discuss it anymore. i know consensus report logic in miner actor.

@whyrusleeping

**Comment by leviok** [comment]
additional, had you keeping watch the chain before 41280?  there was some other issues, but no limit on throughput. Does it exhibit many problems? Are there many forks?

**Comment by SgtCoin** [comment]
Moving to answered due to age and seems the conversation completed as it needed to.
